naga/src/valid/analyzer.rs

```diff
@@ -842,8 +841,16 @@ impl FunctionInfo {
                             && !req.is_empty()
                         {
                             if let Some(cause) = disruptor {
-                                return Err(FunctionError::NonUniformControlFlow(req, expr, cause)
-                                    .with_span_handle(expr, expression_arena));
+                                let severity = DiagnosticFilterNode::search(
+                                    self.diagnostic_filter_head,
+                                    diagnostic_filter_arena,
+                                    FilterableTriggeringRule::DerivativeUniformity,
+                                );
+                                severity.report_diag(
+                                    FunctionError::NonUniformControlFlow(req, expr, cause)
+                                        .with_span_handle(expr, expression_arena),
+                                    |e, level| log::log!(level, "{e}"),
```

Comment:
todo(if-minor): This diagnostic has no source contextualization, and that makes it very difficult for a user to use this warning. Validation errors should be pretty easy to turn into full diagnostics; need to either implement it, if it's easy.

Reply:
Unfortunately, this is not easy, because this context does not have access to source. It's assumed that a user will provide the source when rendering these diagnostics. Punting to <https://github.com/gfx-rs/wgpu/issues/6458>.